### PR TITLE
Added dimm, and memport target support to guard

### PR DESCRIPTION
**Pull request by RameshIyyar**

### CODE REVIEW

File: p10.dts.m4 [hunk 1]
```diff
@@ -260,6 +260,19 @@ define(`id', eval(`$2 % 2'))dnl
...
+
+dnl
+dnl DIMM([index])
+dnl
+define(`DIMM',
+`
+	CONCAT(dimm@, $1) {
+		#address-cells = <0x02>;
+		#size-cells = <0x01>;
+		reg = <0x00 0x08010000 0x3c00>;
```
Comment (ojayanth) [inline p10.dts.m4:273]: This is OCMB specific right? Not sure good for DIMM.

Comment (RameshIyyar) [reply p10.dts.m4:273]: I think so, How do get the address of DIMM to add in DTS? any idea?

Comment (ojayanth) [reply p10.dts.m4:273]: This is oly used for SCOM purposed. you already disabled address translation support , i think you can put default data nnow.

**Comment by amitay** [comment]
For nx and tpm, you need to add the right addresses since they are chiplet level.

**Comment by ojayanth** [comment]
> For nx and tpm, you need to add the right addresses since they are chiplet level.
 Need to provide translate function ,   eg :  .translate = p10_core_translate. There is already hwp for translation in ekb , you can refer.

**Comment by RameshIyyar** [comment]
> For nx and tpm, you need to add the right addresses since they are chiplet level.

Ok, I will push a separate PR for NX and TPM - Updated this PR to just have DIMM and MEM_PORT, please take a look.

**Comment by ojayanth** [comment]
LGTM

**Comment by amitay** [comment]
Merged with changes as discussed separately.

**Pull request closed without merging**